csrc/scheduler/registry.cpp

```diff
@@ -242,7 +242,7 @@ class SchedulerTopologyChecker {
                       // If mapped, and producer is a producer of a reduction,
                       // we can resolve this id
                       ids_to_resolve.erase(
-                          ids_to_resolve.begin() + (entry_i - 1));
+                          ids_to_resolve.begin() + (int64_t)entry_i - 1l);
```

Comment:
This looks ugly to me. Does this warning disappear if the type of `entry_i` is changed to `int`?

Reply:
c++ iterators expect `int64_t`.

```
Error (CLANGTIDY) [bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions,-warnings-as-errors] narrowing conversion from 'size_t' (aka 'unsigned long') to signed type '__gnu_cxx::__normal_iterator<std::pair<nvfuser::IterDomain *, nvfuser::IterDomain *> *, std::vector<std::pair<nvfuser::IterDomain *, nvfuser::IterDomain *>>>::difference_type' (aka 'long') is implementation-defined
```

Reply:
Yes, but does clang-tidy complains when it's `int`?

Reply:
I thought we were sticking to `int64_t`.

Reply:
Yeah, that's right. Is the `l` suffix necessary?

Reply:
Looks like you have already changed to just cast `entry_i`.